experiments/Compute.Experiments/Az.Compute.psm1

```diff
@@ -0,0 +1,331 @@
+
+        Write-Info "Ensuring Vnet...";
+        $vnet = Use-Vnet `
+            -Name $VnetName `
+            -ResourceGroup $ResourceGroup `
+            -Location $Location `
+            -SubnetAddressPrefix $SubnetAddressPrefix `
+            -VnetAddressPrefix $VnetAddressPrefix;
+
+        Write-Info "Ensuring public IP...";
+        $pip = Use-Pip `
+            -Name $PublicIpName `
+            -ResourceGroup $ResourceGroup `
+            -Location $Location `
+            -DnsLabel $PublicIpDnsLabel `
+            -AllocationMethod $PublicIpAllocationMethod `
+            -IdleTimeoutInMinutes $PublicIpIdleTimeoutInMinutes;
+
+        Write-Info "Ensuring NSG...";
+        $nsg = Use-Nsg `
+            -Name $NsgName `
+            -ResourceGroup $ResourceGroup `
+            -Location $Location `
+            -OpenPorts $NsgOpenPorts;
+
+        Write-Info "Ensuring NIC...";
+        $nic = Use-Nic `
+            -Name $NicName `
+            -ResourceGroup $ResourceGroup `
+            -Location $Location `
+            -SubnetId $vnet.Subnets[0].Id `
+            -PublicIpAddressId $pip.Id `
+            -NetworkSecurityGroupId $nsg.Id;
+
+        # TODO: Add disk options (https://docs.microsoft.com/en-us/azure/virtual-machines/scripts/virtual-machines-windows-powershell-sample-create-vm-from-managed-os-disks?toc=%2fpowershell%2fmodule%2ftoc.json)?
+        # https://docs.microsoft.com/en-us/powershell/module/azurerm.compute/set-azurermvmosdisk?view=azurermps-4.2.0
+
+        # Create a virtual machine configuration
+        $vmConfig = New-AzureRmVMConfig -VMName $Name -VMSize  $Size `
+            | Set-AzureRmVMOperatingSystem -Windows -ComputerName $Name -Credential $creds `
+            | Set-AzureRmVMSourceImage `
+                -PublisherName MicrosoftWindowsServer `
+                -Offer WindowsServer `
+                -Skus 2016-Datacenter `
+                -Version latest `
+            | Add-AzureRmVMNetworkInterface -Id $nic.Id
+
+        # Create a virtual machine
+        $vm = New-AzureRmVM -ResourceGroupName $resourceGroup -Location $location -VM $vmConfig
+
+        # Write info about VM.
+
+        # TODO: Remove these and make this a formatter.
+        # Write-Info "$($tab)Resource group: $rgName.";
+        # Write-Info "$($tab)VM Name: MyVm$random.";
+        # Write-Info "$($tab)Location: $location.";
+        # Write-Info "$($tab)FQDN: $fqdn.";
+        Write-Info "$($tab)Username: $username.";
+        Write-Info "$($tab)Password: $ptPassword.";
+
+        return $vm;
+    } catch {
+        Write-Error $_;
+        Write-Error "Something went wrong.  Issue the command again: it is idempotent. :)";
+    }
+}
+
+Export-ModuleMember -Function New-AzVm
+
+# Helpers.
+
+function Use-ResourceGroup {
+    param (
+        [Parameter(Mandatory=$true)] [string] $ResourceGroup,
+        [Parameter(Mandatory=$true)] [string] $Location
+    )
+
+    $rg = Get-AzureRmResourceGroup `
```

Comment:
You will need to use -ErrorAction SilentlyContinue when invoking this, or you will leak a pipeline stopped error.  Fixed in the latest version

Reply:
This applies to all the Get | Select -First calls.

Note- why not use Get-AzureRmResourceGroup -ResourceGroupName instead?  This shoudl return nothign if no resource group is found